src/feedback/mod.rs

```diff
@@ -76,12 +78,31 @@ impl<'a> RtcpPacketParser<'a> for TransportFeedback<'a> {
 }
 
 impl<'a> TransportFeedback<'a> {
-    pub fn builder(fci: impl FciBuilder<'static> + 'static) -> TransportFeedbackBuilder {
+    /// Constructs a [`TransportFeedbackBuilder`] which refers to the provided [`FciBuilder`].
+    ///
+    /// Use [`TransportFeedback::builder_owned`] to return the [`TransportFeedbackBuilder`]
+    /// from a function.
+    pub fn builder(fci: &'a dyn FciBuilder<'a>) -> TransportFeedbackBuilder<'a> {
+        TransportFeedbackBuilder {
+            padding: 0,
+            sender_ssrc: 0,
+            media_ssrc: 0,
+            fci: FciBuilderWrapper::Borrowed(fci),
+        }
+    }
+
+    /// Constructs a [`TransportFeedbackBuilder`] which owns the provided [`FciBuilder`].
+    ///
+    /// This allows returning it from a function.
+    ///
+    /// **Warning:** this causes the [`FciBuilder`] to be heap-allocated.
+    /// Use [`TransportFeedback::builder`] when possible.
+    pub fn builder_owned(fci: impl FciBuilder<'a> + 'static) -> TransportFeedbackBuilder<'a> {
```

Comment:
Instead of the boxing and everything, can we make `TransportFeedbackBuilder` generic over the `FciBuilder` type? Then you can directly store `fci` inside the struct without having to worry about boxing

Reply:
We don't want to propagate the type to the `Compound` type which is a collection of rtcp-types. See the comment: https://github.com/ystreet/rtcp-types/blob/d803ee64ed8124469983417a887a76791ec3d5eb/src/feedback/mod.rs#L402

Reply:
Unfortunate but I don't have a better suggestion either.

We could do some unsafe tricks here with manual vtable handling and storage of the actual value in a `SmallVec` or so to avoid the heap allocation in most cases but that's going to become a bit of code :)